docs/how-to/types.rst

```diff
@@ -0,0 +1,152 @@
+    }
+
+List also implements ``.toString()``, a JavaScript function that is sometimes called automatically when a string
+conversion is needed.
+
+.. code-block:: javascript
+
+    List.prototype.toString = function() {
+        return this.__str__()
+    }
+
+Note also the format for errors: ``throw new exceptions.<Error>.$pyclass``.
+
+Making Changes
+**************
+
+Make a Test
+^^^^^^^^^^^
+
+There is much work to be done in the types folder. When making changes, your goal is to match the output
+of CPython and the output of the same call made in Batavia. Since we know the desired input and output,
+we can use a test and then just fiddle.
+
+Head over to ``/tests`` and find the ``test_<yourtype>`` file. Many types have a robust test suite, but
+do not assume that it is complete. 
+Follow the format there to add a test for your issue or modify an existing test.
+Run it using the following command to check for errors.
+
+.. code-block:: bash
+
+    $ python setup.py -s tests.path.to.your.test.TestClass.test_function
+
+Note: ``@expectedFailure`` indicates a test that's not passing yet. Your issue may be tested in one of those cases already.
+
+Pass the Test
+^^^^^^^^^^^^^
+
+If the test code runs and fails, you've identified the bug and should have some helpful output comparisons. Head over to 
+the type you want and start making edits, running your test until it passes. Occasionally, your bug will take you into 
+other Batavia types and builtins. That's fine too! There are a million places for small omissions all over the codebase.
+Just keep in mind that the further you go down the rabbit hole, the more likely you are to have missed something simple.
+
+Once the test passes, run all tests for the class and see what else broke. (There's always something)::
+
+    $ python setup.py -s tests.path.to.your.test
+
+After that, it's a good idea to pull the upstream master and check for merge conflicts.::
+
+    $ git add .
+    $ git commit -m "<message>"
+    $ git fetch upstream
+    $ git rebase origin/master
+
+If you made major changes, then it may be a good idea to run the full test suite before submitting your pull request.::
+
+    $ python setup.py -s tests
+
+(Check out the sidebar for better/faster ways to run the full suite.) Finally, push your code to your fork and submit
+your pull request on GitHub to run the CI. Fix any issues and push again until CI passes. The Batavia team will get back
+to you with any additional notes and edits.
+
+Tips
+^^^^
+
+Your goal is to mimic the CPython implementation as much as possible. If you do so, you'll often fix multiple issues at once. Here's some tips:
+
+* The original implementation is documented in detail at https://docs.python.org/3/ -- reading up there will definitely improve your understanding.
+* If you're inherting your class from JavaScript, which is very common, you get JavaScript's internal methods for free. Oftentimes, they can be left as is or lightly wrapped.
```

Comment:
@awildtechno Tiny nitpick: `inherting` I think could be `inheriting`?

I'm also not sure if it makes sense to add something referring to tox / tox.ini or https://github.com/beeware/beekeeper (could be in a later PR though)?

Otherwise in my quick overview before sleep, this PR looks awesome and way better than what I just started writing, will definitely enjoy going through it in more depth when PyConAU dev sprints continue tomorrow!